src/Analyzer/QueryNode.cpp

```diff
@@ -185,10 +186,7 @@ void QueryNode::dumpTreeImpl(WriteBuffer & buffer, FormatState & format_state, s
     {
         buffer << '\n' << std::string(indent + 2, ' ') << "SETTINGS";
         for (const auto & change : settings_changes)
-        {
             buffer << fmt::format(" {}={}", change.name, toString(change.value));
-        }
-        buffer << '\n';
```

Comment:
I'm not sure that's right. What will happen in case of union?
```sql
SELECT ...
SETTINGS blabla=1
UNION
SELECT ...
SETTINGS blabla=0
```

Reply:
Query:
```sql
EXPLAIN QUERY TREE (SELECT id FROM test_table SETTINGS optimize_functions_to_subcolumns = 1)
UNION ALL (SELECT id FROM test_table SETTINGS optimize_functions_to_subcolumns = 1);
```
Result before:
```
┌─explain──────────────────────────────────────────────────────────────────────┐
│ UNION id: 0, union_mode: UNION_ALL                                           │
│   QUERIES                                                                    │
│     LIST id: 1, nodes: 2                                                     │
│       QUERY id: 2                                                            │
│         PROJECTION COLUMNS                                                   │
│           id UInt64                                                          │
│         PROJECTION                                                           │
│           LIST id: 3, nodes: 1                                               │
│             COLUMN id: 4, column_name: id, result_type: UInt64, source_id: 5 │
│         JOIN TREE                                                            │
│           TABLE id: 5, table_name: default.test_table                        │
│         SETTINGS optimize_functions_to_subcolumns=1                          │
│                                                                              │
│       QUERY id: 6                                                            │
│         PROJECTION COLUMNS                                                   │
│           id UInt64                                                          │
│         PROJECTION                                                           │
│           LIST id: 7, nodes: 1                                               │
│             COLUMN id: 8, column_name: id, result_type: UInt64, source_id: 9 │
│         JOIN TREE                                                            │
│           TABLE id: 9, table_name: default.test_table                        │
│         SETTINGS optimize_functions_to_subcolumns=1                          │
└──────────────────────────────────────────────────────────────────────────────┘
```

Result after:
```
┌─explain──────────────────────────────────────────────────────────────────────┐
│ UNION id: 0, union_mode: UNION_ALL                                           │
│   QUERIES                                                                    │
│     LIST id: 1, nodes: 2                                                     │
│       QUERY id: 2                                                            │
│         PROJECTION COLUMNS                                                   │
│           id UInt64                                                          │
│         PROJECTION                                                           │
│           LIST id: 3, nodes: 1                                               │
│             COLUMN id: 4, column_name: id, result_type: UInt64, source_id: 5 │
│         JOIN TREE                                                            │
│           TABLE id: 5, table_name: default.test_table                        │
│         SETTINGS optimize_functions_to_subcolumns=1                          │
│       QUERY id: 6                                                            │
│         PROJECTION COLUMNS                                                   │
│           id UInt64                                                          │
│         PROJECTION                                                           │
│           LIST id: 7, nodes: 1                                               │
│             COLUMN id: 8, column_name: id, result_type: UInt64, source_id: 9 │
│         JOIN TREE                                                            │
│           TABLE id: 9, table_name: default.test_table                        │
│         SETTINGS optimize_functions_to_subcolumns=1                          │
└──────────────────────────────────────────────────────────────────────────────┘

```